task_1/contracts/TicketSale.sol

```diff
@@ -0,0 +1,279 @@
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import "@chainlink/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";
+import "@chainlink/contracts/src/v0.8/VRFConsumerBaseV2.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+import "hardhat/console.sol";
+
+interface MyTicket is IERC721 {
+    function batchMint(address to, uint256 amount, string memory uri) external;
+}
+
+contract TicketSale is VRFConsumerBaseV2, ReentrancyGuard, AccessControl {
+    using SafeERC20 for IERC20;
+    using Counters for Counters.Counter;
+
+    MyTicket private NFTAddress;
+    Counters.Counter public currencyId;
+    Counters.Counter public ticketId;
+    uint256 public loteryLimit = 500;
+    uint256 public lotteryFee = 10000; // 1% = 1000
+
+    VRFCoordinatorV2Interface immutable COORDINATOR;
+    uint64 immutable s_subscriptionId;
+    bytes32 immutable s_keyHash;
+    uint32 constant CALLBACK_GAS_LIMIT = 250000;
+    uint32 constant NUM_WORDS = 1;
+    uint16 constant REQUEST_CONFIRMATIONS = 3;
+    uint256 public s_requestId;
+
+    uint256 public priceForOne = 0.02 ether;
+    address payable private owner;
+    uint256 public winnerNumber;
+    uint256 public lotteryDuration = 7 days;
+    uint256 public lotteryStart;
+
+    mapping(uint8 => address) public listOfPiceFeed;
+    mapping(uint8 => address) public listOfToken;
+    mapping(address => bool) public supportOfToken;
+    mapping(uint256 => address) public ownerOfTicket;
+    mapping(uint16 => uint256) public purchaseAmount;
+
+    event PurchaseTicket(
+        address indexed to,
+        uint8 amountOfTicket,
+        uint8 _currencyId,
+        uint256 price
+    );
+    event Withdraw(address indexed to, uint256 amount, uint8 _currencyId);
+    event SetNewToken(address oracleAddress, address tokenAddress, address who);
+    event WinnerNumber(uint256 winner);
+    event Payout(
+        address tokenAddress,
+        address payoutAddress,
+        uint256 amount
+    );
+
+    modifier onlyAdmin() {
+        require(
+            hasRole(DEFAULT_ADMIN_ROLE, msg.sender),
+            "Caller is not a admin"
+        );
+        _;
+    }
+
+    constructor(
+        address _NFTAddress,
+        uint64 subscriptionId,
+        address vrfCoordinator,
+        bytes32 keyHash
+    ) VRFConsumerBaseV2(vrfCoordinator) {
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        NFTAddress = MyTicket(_NFTAddress);
+        owner = payable(msg.sender);
+        COORDINATOR = VRFCoordinatorV2Interface(vrfCoordinator);
+        s_keyHash = keyHash;
+        s_subscriptionId = subscriptionId;
+        listOfPiceFeed[0] = address(0);
```

Comment:
It makes no sense in this case to set zero values, since they are already set there